File: csharp/Exceptions/ApiException.cs
using System;

namespace Authdog.Exceptions
{
    /// <summary>
    /// Exception thrown when API requests fail
    /// </summary>
    public class ApiException : AuthdogException
    {
        public int? StatusCode { get; }

        public ApiException(string message = "API request failed") : base(message)
        {
        }

        public ApiException(string message, Exception innerException) : base(message, innerException)
        {
        }

        public ApiException(string message, int statusCode) : base(message)
        {
            StatusCode = statusCode;
        }
    }
}

File: planned/php/src/Exceptions/ApiException.php
<?php

namespace Authdog\Exceptions;

/**
 * Exception thrown when API requests fail
 */
class ApiException extends AuthdogException
{
    public function __construct(string $message = "API request failed", int $code = 0, ?\Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}
